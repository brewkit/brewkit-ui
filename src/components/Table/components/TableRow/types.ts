import { HTMLProps, ReactNode } from 'react';
import { Props as ColumnProps } from '../TableHeader/components/TableHeaderColumn/types';
import { Props as CellProps } from './components/TableCell/types';


export interface ColumnPropsWithName extends ColumnProps {

    /**
     * The name of the column
     */
    name: string


}


export interface RowData {

    /**
     * Each key is the name of a column and represents a cell in a row
     */
    [key: string]: CellProps | ReactNode
}


export interface Props {


    /**
     * Any custom content to fill the TableRow that is different than default
     */
    children?: ReactNode,


    /**
     * The className for the TableRow
     */
    className?: string,


    /**
     * The configuration that defines the columns in the TableRow
     */
    columnConfig?: ColumnPropsWithName[],


    /**
     * The row data used to populate the TableRow
     */
    rowData?: RowData,


}
