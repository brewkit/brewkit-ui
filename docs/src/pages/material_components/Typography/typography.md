# Typography

## Extends [Typography](https://v4.mui.com/components/typography/#typography)
