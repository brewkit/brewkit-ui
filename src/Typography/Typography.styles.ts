import { createStyles, makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(
    (theme) =>
        createStyles({
            primary: { color: theme.palette.primary.main },
            secondary: { color: theme.palette.secondary.main },
            success: { color: theme.palette.success.main },
            error: { color: theme.palette.error.main },
            warning: { color: theme.palette.warning.main },
            text: { color: theme.palette.text.primary },
        }),
    { name: 'Typography' },
);

export default useStyles;
