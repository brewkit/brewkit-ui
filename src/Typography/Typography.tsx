import React from 'react';
import clsx from 'clsx';
import { merge } from 'lodash';
import {
    Typography as MUITypography,
    TypographyProps as MUITypographyProps,
} from '@material-ui/core';
import withoutKeys from '../utils/withoutKeys';
import useStyles from './Typography.styles';

export interface BkTypographyProps extends Omit<MUITypographyProps, 'color'> {
    color?:
        | 'primary'
        | 'secondary'
        | 'error'
        | 'success'
        | 'warning'
        | 'textPrimary'
        | 'textSecondary';
}

const Typography = React.forwardRef(
    (props: BkTypographyProps, ref: React.Ref<HTMLElement>) => {
        const {
            children,
            className,
            classes: userClasses = {},
            color = 'textPrimary',
            variant = 'body1',
            ...otherProps
        } = props;
        const typographyClasses = useStyles();
        const classes = merge(typographyClasses, userClasses);
        const muiClasses = withoutKeys(classes, [
            'error',
            'primary',
            'secondary',
            'success',
            'textPrimary',
            'textSecondary',
            'warning',
        ]);

        return (
            <MUITypography
                className={clsx(className, classes?.[color])}
                classes={muiClasses}
                color={undefined} // We manually resolve colors color
                ref={ref}
                variant={variant}
                {...otherProps}>
                {children}
            </MUITypography>
        );
    },
);

Typography.displayName = 'BkTypography';

export default Typography;
