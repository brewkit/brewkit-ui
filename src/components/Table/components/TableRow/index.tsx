import React, { ReactElement, ReactNode } from 'react';
import clsx from 'clsx';
import TableCell from './components/TableCell';
import { Props } from './types';


/**
 * The `<TableRow />` component
 */
function TableRow({
    className,
    children,
    columnConfig,
    rowData,
    ...otherProps
}: Props): ReactElement {


    const classes = clsx(
        'brew-TableRow',
        className,
    );


    const content = children || (columnConfig?.map((col, index) => {
        if (rowData && col?.name && rowData[col.name]) {
            const cellData = rowData[col.name];
            if (cellData && typeof cellData === 'object' && 'label' in cellData) {
                return <TableCell key={index} {...cellData} />;
            }
            return (
                <TableCell key={index}>
                    {cellData}
                </TableCell>
            );
        }
    }));

    
    return (
        <tr className={classes} {...otherProps}>
            {content}
        </tr>
    );


}


TableRow.Cell = TableCell;


export default TableRow;
