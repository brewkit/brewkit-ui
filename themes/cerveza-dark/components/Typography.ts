

import { color } from '../../cerveza/variables';


const Typography = {


    overrides: {

        MuiTypography: {

            subtitle1: {
                color: color.blue.light,
            },

            subtitle2: {
                color: color.blue.light,
            },

            button: {
                color: color.blue.main,
            },

            body1: {
                color: color.blueGray.light,
            },

            body2: {
                color: color.blueGray.light,
            },

            caption : {
                color: color.blueGray.light,
            }

        },

    },


};


export default Typography;
