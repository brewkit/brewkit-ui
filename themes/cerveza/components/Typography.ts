import { color } from '../variables';


const Typography = {


    typography: {
        fontSize: 16,
        h1: {
            fontWeight: 'bold',
            fontSize: '5rem',
            lineHeight: 1.2,
        },
        h2: {
            fontWeight: 'bold',
            fontSize: '3rem',
            lineHeight: 1.25,
        },
        h3: {
            fontWeight: 'bold',
            fontSize: '2rem',
            lineHeight: 1.5,
        },
        h4: {
            fontWeight: 'bold',
            fontSize: '1.5rem',
            lineHeight: 1.33,
        },
        h5: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        h6: {},
        body1: {
            lineHeight: 1.25,
        },
        body2: {
            lineHeight: 1.25,
        },
        subtitle1: {
            fontWeight: 'bold',
            fontSize: '1.5rem',
            lineHeight: 1.33,
        },
        subtitle2: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        button: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        caption: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
        overline: {
            fontWeight: 'bold',
            fontSize: '1.25rem',
            lineHeight: 1.2,
        },
    },


    overrides: {

        MuiTypography: {

            subtitle1: {
                color: color.blue.dark,
            },

            subtitle2: {
                color: color.blue.dark,
            },

            button: {
                color: color.blue.main,
            },

            body1: {
                color: color.blueGray.dark,
            },

            body2: {
                color: color.blueGray.dark,
            },

        },

    },


};


export default Typography;
