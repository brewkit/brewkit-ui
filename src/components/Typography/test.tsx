import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Typography from './index';


it('renders correctly', () => {
    const tree = renderer
        .create(<Typography>Foo</Typography>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders children when passed in', () => {
    const wrapper = shallow((
        <Typography>
            <span>foo</span>
        </Typography>
    ));
    expect(wrapper.contains(<span>foo</span>)).toEqual(true);
});


it('accepts custom classes', () => {
    const wrapper = shallow((
        <Typography className="customClass">Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('customClass')).toEqual(true);
    expect(wrapper.hasClass('brew-Typography')).toEqual(true);
});


it('"variant" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--variant-inherit')).toEqual(true);
});


it('"color" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--color-initial')).toEqual(true);
});


it('"display" defaults to "initial" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--display-initial')).toEqual(true);
});


it('"align" defaults to "inherit" and applies appropriate class', () => {
    const wrapper = shallow((
        <Typography>Lorem ipsum dolor sit amet.</Typography>
    ));
    expect(wrapper.hasClass('brew-Typography--align-inherit')).toEqual(true);
});
