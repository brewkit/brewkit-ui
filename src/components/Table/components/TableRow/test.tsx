import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import TableRow from './index';
import Typography from '../../../Typography';


describe('Table Row', () => {


    it('Renders correctly with children', () => {
        const content = renderer
            .create((
                <TableRow>
                    <td>Column1</td>
                    <td>Column2</td>
                    <td>Column3</td>
                </TableRow>
            ))
            .toJSON();

        expect(content).toMatchSnapshot();
    });


    it('Includes custom className', () => {
        const className = `class-${String(Math.floor(Math.random() * 1000))}`;
        const content = shallow((
            <TableRow className={className}>
                <td>Column4</td>
                <td>Column5</td>
                <td>Column6</td>
            </TableRow>
        ));

        expect(content.hasClass(className)).toBe(true);
        expect(content.hasClass('brew-TableRow')).toBe(true);
    });


});

describe('Table Row Props', () => {


    it('Renders `columnConfig` and `rowData` props correctly', () => {
        const content = renderer
            .create((
                <TableRow
                    columnConfig={[
                        { name: 'ColA' },
                        { name: 'ColB' },
                        { name: 'ColC' },
                    ]}
                    rowData={{
                        ColA: 'ColumnA',
                        ColB: 'ColumnB',
                        ColC: 'ColumnC',
                    }}
                />
            ))
            .toJSON();

        expect(content).toMatchSnapshot();

        expect(shallow((
            <TableRow
                columnConfig={[
                    { name: 'ColD' },
                    { name: 'ColE' },
                    { name: 'ColF' },
                ]}
                rowData={{
                    ColD: <Typography variant="body1">ColumnD</Typography>,
                    ColE: <Typography variant="body1">ColumnE</Typography>,
                    ColF: <Typography variant="body1">ColumnF</Typography>,                    
                }}
            />
        ))
            .find(Typography)
            .find('[variant="body1"]')
            .children()
            .contains('ColumnE')).toBe(true);
    });


});
