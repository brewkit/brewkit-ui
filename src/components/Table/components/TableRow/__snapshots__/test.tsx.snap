// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row Props Renders \`columnConfig\` and \`rowData\` props correctly 1`] = `
<tr
  className="brew-TableRow"
>
  <td
    className="brew-TableCell"
  >
    ColumnA
  </td>
  <td
    className="brew-TableCell"
  >
    ColumnB
  </td>
  <td
    className="brew-TableCell"
  >
    ColumnC
  </td>
</tr>
`;

exports[`Table Row Renders correctly with children 1`] = `
<tr
  className="brew-TableRow"
>
  <td>
    Column1
  </td>
  <td>
    Column2
  </td>
  <td>
    Column3
  </td>
</tr>
`;
